server/src/graql/internal/reasoner/atom/AtomicFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * GRAKN.AI - THE KNOWLEDGE GRAPH
+ * Copyright (C) 2018 Grakn Labs Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package grakn.core.graql.internal.reasoner.atom;
+
+import grakn.core.graql.admin.Atomic;
+import grakn.core.graql.admin.ReasonerQuery;
+import grakn.core.graql.internal.executor.property.PropertyExecutor;
+import grakn.core.graql.query.pattern.Conjunction;
+import grakn.core.graql.query.pattern.Statement;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Factory class for creating {@link Atomic} objects.
+ */
+public class AtomicFactory {
+
+    /**
+     * @param pattern conjunction of patterns to be converted to atoms
+     * @param parent query the created atoms should belong to
+     * @return set of atoms
+     */
+    public static Stream<Atomic> createAtoms(Conjunction<Statement> pattern, ReasonerQuery parent) {
```

Comment:
Why did you extract this method to its own class again @kasper-piskorski ? I moved it into `ReasonerQueryImpl` because it's one method made only used by that class, and it doesn't need its own class. I'm trying to minimise the number of classes that don't carry significant weight/purpose.

Reply:
Adding more things later.

Reply:
Okay, you should mention that (e.g. in the description) and tag relevant person (in this case me), otherwise it looks _rather impolite_ to just undo the change that someone just made in the previous PR. I'll approve this PR once my other PR is in.